tests/component/system/test_device.py

```diff
@@ -37,3 +37,7 @@ def test___self_test_device___no_errors(any_x_series_device: Device) -> None:
 
 def test___self_cal___no_errors(any_x_series_device: Device) -> None:
     any_x_series_device.self_cal()
+
+
+def test___restore_last_ext_cal_const___no_errors(any_x_series_device: Device) -> None:
```

Comment:
Reorder these tests so it runs this one first and self-cal second. Otherwise running the nidaqmx-python tests leaves one of your X Series devices with old calibration constants.

Reply:
or change these tests to use `sim_x_series_device` to leave the cal constants alone on any real X Series devices

Reply:
`self_cal` and `restore_last_ext_cal_const` test cases are updated to test with simulated device.